server/src/server.ts

```diff
@@ -207,7 +213,15 @@ function openProjectFile(workspacePath: string, fileUri: string): boolean {
                         line.search(GENERATED_YML_DIR_REGEX) === -1
                     );
                 })
-                .map((line) => path.join(path.dirname(fileUri), line))
+                .map((line) => {
+                    if (isTypeM) {
+                        // In a M type *.kao file, paths are relative to the current *.kao file.
+                        return path.join(path.dirname(fileUri), line);
+                    }
+                    // In a F type *.kao file, paths are relative to the project's root.
+                    // Here we assume that the root is the workspace path.
+                    return path.join(workspacePath, line);
```

Comment:
Might be slightly nicer with an `else` to highlight the M / F opposition.

Reply:
OK but it doesn't seem that much of an improvement since there are just a few lines of code + comments.

Reply:
In one case you’ve got a clear “this or that”, and in the other (the current version) it looks like F is a garbage-collecting case or something xD.

Reply:
Besides, those comments about the ≠ between F and M would have made a nice PR description :cry:

Reply:
My linter didn't let me add this useless `else`.
I used the `cond ? a : b;` operator instead.

Reply:
LOL What was the explanation? And that’s called the “ternary” operator xd